source/parser.hera

```diff
@@ -3637,10 +3664,20 @@ CaseExpressionList
 
     return result
 
+CaseExpression
+  # NOTE: Explicitly matching `case foo:` to prevent this from looking like
+  # an object literal
+  PropertyName &( _? Colon ) -> $1
```

Comment:
I tried doing this, but then `case [foo]: bar` got matched as `case {[foo]: bar}` which seems slightly worse than `case ([foo]): bar`. Instead I now look for `ComputedProperty` and unwrap if it's "implicit" (template literal or negative number) and otherwise recast to `ArrayExpression`.